### PR TITLE
Fix https://github.com/eclipse/smarthome/issues/3912 in CronExpression

### PR DESCRIPTION
**Pull request by kgoderis**
Signed-off-by: Karel Goderis <karel.goderis@me.com>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/CronExpression.java [hunk 1]
```diff
@@ -223,16 +225,29 @@ public static boolean isValidExpression(String cronExpression) {
...
+
+        if (yearsPart == null) {
+            List<@NonNull CronExpressionPart> ep = getExpressionParts();
+            List<@NonNull CronExpressionPart> parts = new LinkedList<@NonNull CronExpressionPart>();
+            parts.addAll(ep);
+
+            try {
+                parts.add(new YearsExpressionPart("*"));
+            } catch (ParseException e) {
+                throw new IllegalArgumentException("Year rule part can contain * as a token");
```
Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/CronExpression.java]: ~~can~~ -> cannot

**Comment by kgoderis** [comment]
> How about a little regression test in CronExpressionTest?

I am not familiar with how to craft these kind of tests....

**Comment by kaikreuzer** [comment]
@kgoderis We already have a test for this and you have removed the "Ignore" from it - so maybe it would suffice to also [remove the FIXME comment](https://github.com/eclipse/smarthome/pull/4215/files#diff-90461fad9ab4c7bd9de458243afdb5bcR81)?

**Comment by kgoderis** [comment]
done.

**Pull request merged**